tests/tests.rs

```diff
@@ -32,6 +32,17 @@ fn wait_for_server(addr: &str) {
     panic!("failed to connect to {:?} after 10 tries", addr);
 }
 
+fn curl_version() -> String {
+    String::from_utf8_lossy(
+        &Command::new("curl")
+            .arg("--version")
+            .output()
+            .expect("cannot run curl to collect --version")
+            .stdout,
+    )
+    .to_string()
+}
```

Comment:
Style nits: I'd prefer `into_owned()` on the `Cow` at the end instead of `to_string()`; in this case, we don't need the full power of `Display`, so using a more specific method makes it clearer what's going on.

Also, I usually prefer not to add small single-use functions, since I find that the indirection makes it harder to understand what's going rather than easier.

Reply:
Resolved in ffa7677285ff1758f31e6dbc04118b0bde9d43f7